src/main/host/thread_ptrace.c

```diff
@@ -715,31 +713,28 @@ Thread* threadptrace_new(Host* host, Process* process, gint threadID) {
     ThreadPtrace* thread = g_new(ThreadPtrace, 1);
 
     *thread = (ThreadPtrace){
-        .base = (Thread){.run = threadptrace_run,
-                         .resume = threadptrace_resume,
-                         .terminate = threadptrace_terminate,
-                         .getReturnCode = threadptrace_getReturnCode,
-                         .isRunning = threadptrace_isRunning,
-                         .free = threadptrace_free,
-                         .newClonedPtr = threadptrace_newClonedPtr,
-                         .releaseClonedPtr = threadptrace_releaseClonedPtr,
-                         .getReadablePtr = threadptrace_getReadablePtr,
-                         .getReadableString = threadptrace_getReadableString,
-                         .getWriteablePtr = threadptrace_getWriteablePtr,
-                         .flushPtrs = threadptrace_flushPtrs,
-                         .nativeSyscall = threadptrace_nativeSyscall,
-
-                         .type_id = THREADPTRACE_TYPE_ID,
-                         .referenceCount = 1},
+        .base = thread_create(host, process, THREADPTRACE_TYPE_ID,
+                              (ThreadMethods){
+                                  .run = threadptrace_run,
+                                  .resume = threadptrace_resume,
+                                  .terminate = threadptrace_terminate,
+                                  .getReturnCode = threadptrace_getReturnCode,
+                                  .isRunning = threadptrace_isRunning,
+                                  .free = threadptrace_free,
+                                  .newClonedPtr = threadptrace_newClonedPtr,
+                                  .releaseClonedPtr = threadptrace_releaseClonedPtr,
+                                  .getReadablePtr = threadptrace_getReadablePtr,
+                                  .getReadableString = threadptrace_getReadableString,
+                                  .getWriteablePtr = threadptrace_getWriteablePtr,
+                                  .flushPtrs = threadptrace_flushPtrs,
+                                  .nativeSyscall = threadptrace_nativeSyscall,
+                              }),
         // FIXME: This should the emulated CPU's frequency
         .tsc = {.cyclesPerSecond = 2000000000UL},
```

Comment:
Indentation again. I think it might be because the following line was not edited so clang-format isn't able to keep the entire initializer consistent?

Reply:
Same here